File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        	int n = (int)numbers.size();
	int l = 0 , r = n - 1;
	vector<int> ans;

	while(l < r){
		int sum = numbers[l] + numbers[r];

		if(sum > target){
			r --;
		}
		else if (sum < target){
			l ++;
		}

		else {
			ans = {l+1 , r+1};
			break;
		}

	}

	return ans;

        
        
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
            l, r = 0, len(numbers) - 1
            while l < r :
                t_sum = numbers[l] + numbers[r]

                if t_sum == target:
                    return [l+1, r+1]
                elif t_sum > target:
                    r -= 1
                else:
                    l+=1
